src/conn/conn_api.c

```diff
@@ -1645,9 +1708,11 @@ __conn_hash_config(WT_SESSION_IMPL *session, const char *cfg[])
     /* Don't set the values in the statistics here. They're set after the connection is set up. */
 
     /* Hash bucket arrays. */
+    WT_RET(__wt_calloc_def(session, conn->hash_size, &conn->bkuphash));
```

Comment:
I think this is OK here, but it feels like a very general place to put the hash bucket and to always allocate one. Did you think about instead passing the `cfg` to `__wt_turtle_init` and keeping the data structures local to the backup code in there?
It looks like you need to create and save an associated ID list for RTS, and that probably needs to belong in the WT_CONNECTION (since it's shared across subsystems), but the hash table here appears more localised.

Reply:
I have done the initialisation and destruction of the hashtable inside `__wt_turtle_init` instead now. I have maintained the backup hash in the connection structure still. The reason is because we will need to access at multiple mounts across different functions.